patronload/patron.py

```diff
@@ -62,6 +67,152 @@ def populate_patron_common_fields(
+            split_name[1].strip() or ""
+        )
+
+    patron_template.user_group.string = (  # type: ignore[union-attr]
+        patron_dict["LIBRARY_PERSON_TYPE_CODE"] or ""
+    )
+    patron_template.user_group["desc"] = (  # type: ignore[index,union-attr]
+        patron_dict["LIBRARY_PERSON_TYPE"] or ""
+    )
+
+    patron_template.line1.string = (  # type: ignore[union-attr]
+        patron_dict["OFFICE_ADDRESS"]
+        if patron_dict["OFFICE_ADDRESS"]
+        else "NO ADDRESS ON FILE IN DATA WAREHOUSE"
+    )
+    if patron_dict["OFFICE_PHONE"]:
+        patron_template.find_all("phone_number")[0].string = format_phone_number(
+            patron_dict["OFFICE_PHONE"],
+        )
+    else:
+        patron_template.phones.clear()  # type: ignore[union-attr]
+
+    if patron_dict["ORG_UNIT_ID"] in STAFF_DEPARTMENTS:
+        patron_template.statistic_category.string = (  # type: ignore[union-attr]
+            STAFF_DEPARTMENTS[patron_dict["ORG_UNIT_ID"]]
+        )
+    else:
+        patron_template.statistic_category.string = "ZQ"  # type: ignore[union-attr]
+        logger.error(
+            "Unknown dept: '%s' in record with MIT ID # '%s'",
+            patron_dict["ORG_UNIT_ID"],
+            patron_dict["MIT_ID"],
+        )
+    patron_template.statistic_category["desc"] = (  # type: ignore[index,union-attr]
+        patron_dict["ORG_UNIT_TITLE"] or "Unknown"
+    )
+    return patron_template
+
+
+def populate_student_fields(
+    patron_template: BeautifulSoup,
+    patron_dict: dict[str, Any],
+) -> BeautifulSoup:
+    """
+    Populate the student fields in a patron record.
+
+    The order of the patron record fields must be in same order as they in the
+    STUDENT_FIELDS value list.
+
+    Args:
+        patron_template: An XML template for a patron record
+        patron_dict: A dict of patron record values.
+    """
+    patron_template.first_name.string = (  # type: ignore[union-attr]
+        patron_dict["FIRST_NAME"] or ""
+    )
+    patron_template.middle_name.string = (  # type: ignore[union-attr]
+        patron_dict["MIDDLE_NAME"] or ""
+    )
+    patron_template.last_name.string = (  # type: ignore[union-attr]
+        patron_dict["LAST_NAME"] or ""
+    )
+
+    patron_template.address.line1.string = (  # type: ignore[union-attr]
+        patron_dict["TERM_STREET1"]
+        if patron_dict["TERM_STREET1"]
+        else "NO ADDRESS ON FILE IN DATA WAREHOUSE"
+    )
+    patron_template.line3.string = (  # type: ignore[union-attr]
+        patron_dict["TERM_STREET2"] or ""
+    )
+    patron_template.city.string = (  # type: ignore[union-attr]
+        patron_dict["TERM_CITY"] or ""
+    )
+    patron_template.state_province.string = (  # type: ignore[union-attr]
+        patron_dict["TERM_STATE"] or ""
+    )
+    patron_template.postal_code.string = (  # type: ignore[union-attr]
+        patron_dict["TERM_ZIP"] or ""
+    )
+
+    office_phone_number = patron_template.find_all("phone_number")[0]
```

Comment:
This is an involved code block for the phone numbers so I would welcome any suggestions to simplify it

Reply:
This might make the intent a bit clearer?
```
preferred_phone_number = patron_template.find_all("phone_number")[0]
preferred_phone_number.string = format_phone_number(
    patron_dict["OFFICE_PHONE"]
    or patron_dict["TERM_PHONE1"]
    or patron_dict["TERM_PHONE2"]
    or ""
)
if patron_dict["OFFICE_PHONE"]:
    other_phone_number = patron_template.find_all("phone_number")[1]
    other_phone_number.string = format_phone_number(
        patron_dict["TERM_PHONE1"] or ""
    )
for phone in patron_template.find_all("phone"):
    if not phone.phone_number.string:
        phone.extract()
```